barretenberg/cpp/src/barretenberg/stdlib_circuit_builders/goblin_ultra_flavor.hpp

```diff
@@ -427,6 +427,108 @@ class GoblinUltraFlavor {
                 commitment = proving_key.commitment_key->commit(polynomial);
             }
         }
+        VerificationKey(const size_t circuit_size,
+                        const size_t num_public_inputs,
+                        const size_t pub_inputs_offset,
+                        const Commitment& q_m,
+                        const Commitment& q_c,
+                        const Commitment& q_l,
+                        const Commitment& q_r,
+                        const Commitment& q_o,
+                        const Commitment& q_4,
+                        const Commitment& q_arith,
+                        const Commitment& q_delta_range,
+                        const Commitment& q_elliptic,
+                        const Commitment& q_aux,
+                        const Commitment& q_lookup,
+                        const Commitment& q_busread,
+                        const Commitment& q_poseidon2_external,
+                        const Commitment& q_poseidon2_internal,
+                        const Commitment& sigma_1,
+                        const Commitment& sigma_2,
+                        const Commitment& sigma_3,
+                        const Commitment& sigma_4,
+                        const Commitment& id_1,
+                        const Commitment& id_2,
+                        const Commitment& id_3,
+                        const Commitment& id_4,
+                        const Commitment& table_1,
+                        const Commitment& table_2,
+                        const Commitment& table_3,
+                        const Commitment& table_4,
+                        const Commitment& lagrange_first,
+                        const Commitment& lagrange_last,
+                        const Commitment& lagrange_ecc_op,
+                        const Commitment& databus_id)
+        {
+            this->circuit_size = circuit_size;
+            this->log_circuit_size = numeric::get_msb(this->circuit_size);
+            this->num_public_inputs = num_public_inputs;
+            this->pub_inputs_offset = pub_inputs_offset;
+            this->q_m = q_m;
```

Comment:
not sure how to avoid these copys, but also not sure if its that meaningful for performance

Reply:
I'm not worried about the copies but these big explicit lists are not ideal - Is there no way to use the PrecomputedEntities notion here? Can we construct a PrecomputedEntities from the deserialized data then loop to set the internals? If you can't figure something out can you leave a comment or TODO explaining that its exactly the PrecomputedEntities list but there's just no nice way to use that fact? Otherwise someone might spend time combing through for a subtle difference.

BTW it wasn't obvious to me where this constructor being used, is it here: `std::make_shared<VerificationKey>(from_buffer<VerificationKey>(read_file(vk_path)));`?

Reply:
This constructor is used by MsgPack I believe. It gave me an error without this constructor, but it makes sense because it should call some constructor when it tries to deserialize